src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,46 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User can't be null");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("User's login can't be null");
+        }
+        if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new RegistrationException("User's login must have at least "
+                    + MIN_LOGIN_LENGTH + " characters");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("User's password can't be null");
+        }
+        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RegistrationException("User's password must have at least "
+                    + MIN_PASSWORD_LENGTH + " characters");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("User's age can't be null");
+        }
+        if (user.getAge() < MIN_AGE) {
+            throw new RegistrationException("User's age can't be less than "
+                    + MIN_AGE);
+        }
```

Comment:
The check for age is correct. Ensure that the exception message is informative and specifies the minimum age required.